### ISSUE TITLE
Update / fix dependencies and update .gitignore for user configs

### ISSUE THREAD

**Issue by f4ww4z** [DSC-Galgotias/TechnoJam-Chat#31]
We should keep 3rd party libraries up-to-date.

Also, there is an error when building this project:
```
Warning: The app gradle file must have a dependency on com.google.firebase:firebase-core for Firebase services to work as intended.
```
which looks like a self-explanatory fix.

Next, all `compile` commands will be obsolete, and should be replaced with `implementation` and `api` instead.

Finally, update `.gitignore` to remove build files & other user configs.

**Comment by honeycoder96** [comment]
Hi @f4ww4z ,
Updating the firebase dependencies creates an error as the way we fetch images from the firebase storage has been changed in the latest versions.
so if someone wants to contribute to this issue please make sure that after updating the dependencies you update the UserActivity, ChatFragment and other places which deal with fetching the images from the firebase storage.

Thank you.

**Comment by f4ww4z** [comment]
I'd like to take this issue first before doing #30 . Just to clarify: do we submit PRs in this repo's `development` branch or in the [base repo's](https://github.com/honeycoder96/TechnoJam-Chat/tree/development)

**Comment by honeycoder96** [comment]
hi @f4ww4z .
You can work on this issue.

And yes all the pull requests are to be made in this repository. (DSC-Galgotias)

**Comment by honeycoder96** [comment]
This issue has been solved in #32 .
